specs/networking/p2p-interface.md

```diff
@@ -382,21 +394,29 @@ Request Content
 Response Content:
 ```
 (
-  blocks: []BeaconBlock
+  []BeaconBlock
 )
 ```
 
 Requests count beacon blocks from the peer starting from `start_slot` on the chain defined by `head_block_root`. The response MUST contain no more than count blocks. `step` defines the slot increment between blocks. For example, requesting blocks starting at `start_slot` 2 with a step value of 2 would return the blocks at [2, 4, 6, …]. In cases where a slot is empty for a given slot number, no block is returned. For example, if slot 4 were empty in the previous example, the returned array would contain [2, 6, …]. A step value of 1 returns all blocks on the range `[start_slot, start_slot + count)`.
 
-`BeaconBlocks` is primarily used to sync historical blocks.
+The request MUST be encoded as an SSZ-container.
+
+The response MUST consist of at least one `response_chunk` and MAY consist of many. Each _successful_ `response_chunk` MUST contain a single `BeaconBlock` payload.
+
+`BeaconBlocksByRange` is primarily used to sync historical blocks.
 
 Clients MUST support requesting blocks since the start of the weak subjectivity period and up to the given `head_block_root`.
 
 Clients MUST support `head_block_root` values since the latest finalized epoch.
 
-#### RecentBeaconBlocks
+Clients MUST respond with at least one block, if they have it.
+
+Clients MUST order blocks by increasing slot number.
+
+#### BeaconBlocksByRoot
 
-**Protocol ID:** `/eth2/beacon_chain/req/recent_beacon_blocks/1/`
+**Protocol ID:** `/eth2/beacon_chain/req/beacon_blocks_by_root/1/`
 
 Request Content:
```

Comment:
shouldn't both the request and response be a list of roots/blocks rather than a container ? 
Request Content:
```

(
  []HashTreeRoot
)
```
Response Content:
```
(
 []BeaconBlock
)
```

Reply:
Looks like it